src/azure.eventhub/client.bal

```diff
@@ -0,0 +1,378 @@
+            if (xmlPayload is xml) {
+                return xmlPayload;
+            }
+            return error(EVENT_HUB_ERROR, message = "invalid response while revoking publisher: " + publisherName
+                + ". " + response.statusCode.toString());
+        } else {
+            return error(EVENT_HUB_ERROR, message = "error invoking EventHub API ", cause = response);
+        }
+    }
+
+    # Resume a publisher
+    #
+    # + publisherName - publisher name 
+    # + return - Return publisher details or error
+    public remote function resumePublisher(string publisherName) returns @tainted xml|error {
+        http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
+        http:Request req = new;
+        string token = self.getSASToken();
+        req.addHeader("Authorization", token);
+        var response = clientEndpoint->delete("/revokedpublishers/" + publisherName, req);
+        if (response is http:Response) {
+            var xmlPayload = response.getXmlPayload();
+            if (xmlPayload is xml) {
+                return xmlPayload;
+            }
+            return error(EVENT_HUB_ERROR, message = "invalid response from EventHub API " + response.statusCode.toString());
+        } else {
+            return error(EVENT_HUB_ERROR, message = "error invoking EventHub API ", cause = response);
+        }
+    }
+
+        # Lit available partitions
+        #
+        # + consumerGroupName - consumer group name
+        # + return - Return partition list or error
+        public remote function listPartitions(string consumerGroupName) returns @tainted xml | error {
+            http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
+            http:Request req = new;
+            string token = self.getSASToken();
+            req.addHeader("Authorization", token);
+            var response = clientEndpoint->get("/consumergroups/" + consumerGroupName + "/partitions", req);
+            if (response is http:Response) {
+                var xmlPayload = response.getXmlPayload();
+                if (xmlPayload is xml) {
+                    return xmlPayload;
+                }
+                return error(EVENT_HUB_ERROR, message = "invalid response from EventHub API " + response.statusCode.toString());
+            } else {
+                return error(EVENT_HUB_ERROR, message = "error invoking EventHub API ", cause = response);
+            }
+        }
+
+        # Get partition details
+        #
+        # + consumerGroupName - consumer group name
+        # + partitionId - partitionId
+        # + return - Returns partition details
+        public remote function getPartition(string consumerGroupName, int partitionId) returns @tainted xml | error {
+            http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
+            http:Request req = new;
+            string token = self.getSASToken();
+            req.addHeader("Authorization", token);
```

Comment:
Can we move this to a separate function if it makes sense. It looks duplicated in most remove functions.